packages/origin-247-transfer/src/EnergyTransferRequest.ts

```diff
@@ -0,0 +1,157 @@
+import { ValidateTransferCommandCtor } from './commands/ValidateTransferCommand';
+
+export enum TransferValidationStatus {
+    Pending = 'pending',
+    Valid = 'valid',
+    Invalid = 'invalid',
+    Error = 'error'
+}
+
+export interface EnergyTransferRequestPublicAttrs {
+    id: number;
+    generatorId: string;
+    sellerAddress: string;
+    buyerAddress: string;
+    volume: string;
+    transferDate: Date;
+}
+
+export interface EnergyTransferRequestAttrs extends EnergyTransferRequestPublicAttrs {
+    createdAt: Date;
+    updatedAt: Date;
+
+    certificateId: number | null;
+    isCertificatePersisted: boolean;
+
+    validationStatusRecord: Record<string, TransferValidationStatus>;
+
+    /**
+     * This property is computed on `validationStatusRecord` update
+     * It can be used for queries
+     */
+    computedValidationStatus: TransferValidationStatus;
+}
+
+interface NewAttributesParams {
+    buyerAddress: string;
+    sellerAddress: string;
+    volume: string;
+    generatorId: string;
+    transferDate: Date;
+}
+
+export class EnergyTransferRequest {
+    private constructor(private attrs: EnergyTransferRequestAttrs) {}
```

Comment:
`attrs` approach is what I (and my teammates) have already tested thouroughly on other projects. It's really useful